### ISSUE TITLE
new

### ISSUE THREAD

**Issue by OlgaOrel** [OlgaOrel/pandas#1]
echo "# pandas" >> README.md
git init
git add README.md
git commit -m "first commit"
git remote add origin https://github.com/OlgaOrel/pandas.git
git push -u origin master

**Comment by OlgaOrel** [comment]
import pandas as pd
import numpy as np
from colorama import Fore

k=1
N=100

df = pd.DataFrame({'A': np.random.randint(k, k+1000, size = N),
                   'B': np.random.randint(k, k+10, size = N)
                   })
df['C']= round(df['A']**2/df['B'])
print(df.head())

df['A_1'] = (df['A']<500)*1
print(df.head())
df.loc[df['A_2']<500, 'A'] = 500
print(df.head())

# data = pd.get_dummies(data, columns=["x2", "x3", "x4", "x11", "x15"]) - преобразовываем из категориальных переменных в бинарные

# запускаем обычную линейную регрессию с функцией потерь huber и оптимизируем её через стохастический градиентный спуск.

from sklearn.linear_model import SGDRegressor
sgd = SGDRegressor(loss='huber', n_iter=100)
sgd.fit(train, target)
test  = np.hstack((test, sgd.predict(test)[None].T))
train = np.hstack((train, sgd.predict(train)[None].T))

#Сначала я взял k=7, потом добавил 3 и 15 (в два раза больше и в два раза меньше). Предсказания каждого из этих алгоритмов —
# номера кластеров для каждого образца. Так как эти номера не упорядочены, то оставить их числами нельзя, надо обязательно бинаризовать.

from sklearn.cluster import KMeans
k15 = KMeans(n_clusters=15, precompute_distances = True, n_jobs=-1)
k15.fit(train)
k7 = KMeans(n_clusters=7, precompute_distances = True, n_jobs=-1)
k7.fit(train)
k3 = KMeans(n_clusters=3, precompute_distances = True, n_jobs=-1)
k3.fit(train)
test  = np.hstack((test,  k15.predict(test)[None].T,  k7.predict(test)[None].T,  k3.predict(test)[None].T))
train = np.hstack((train, k15.predict(train)[None].T, k7.predict(train)[None].T, k3.predict(train)[None].T))


#Прежде чем начинать собственно бустить по-настоящему и получать отличные результаты, я решил посмотреть,
# насколько действительно важны все те колонки, которые были даны, и те, которые насоздавал я,
# развернув хэши и проведя кластеризацию методом К-средних. Для этого я провел легкий бустинг (не очень много деревьев),
# и построил столбцы, отсортированные по важности (по мнению xgboost).

gbm = xgb.XGBClassifier(silent=False, nthread=4, max_depth=10, n_estimators=800, subsample=0.5, learning_rate=0.03, seed=1337)
gbm.fit(train, target)
bst = gbm.booster()
imps = bst.get_fscore()

**Comment by OlgaOrel** [comment]
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
plt.style.use('ggplot')
import seaborn as sns
plt.rcParams['figure.figsize'] = (10, 8)

data  = pd.read_csv('https://raw.githubusercontent.com/Yorko/mlcourse_open/master/data/adult.data.csv', sep = ',')
print  (data.head())

#1. Сколько мужчин и женщин (признак sex) представлено в этом наборе данных?
print('#1')
print(data['sex'].value_counts())

#2. Каков средний возраст (признак age) женщин?
print('#2')
print(data[data['sex'] == 'Female']['age'].mean())

#3. Какова доля граждан Германии (признак native-country)?
print('#3')
n3 = pd.DataFrame(data['native-country'].value_counts(normalize = True))
print(n3.loc['Germany'])

#4. Постройте гистограмму распределения (bar plot) образования людей (признак education).
print('#4')
n4 = pd.DataFrame(data['education'].value_counts(normalize = True))
n4.plot(kind = 'bar')
#plt.show()

#5. Каковы средние значения и среднеквадратичные отклонения возраста тех, кто получает более 50K в год (признак salary)
# и тех, кто получает менее 50K в год?
print('#5')
print(data[data['salary'] == '<=50K']['age'].mean())
print(data[data['salary'] == '<=50K']['age'].std())
print(data[data['salary'] != '<=50K']['age'].mean())
print(data[data['salary'] != '<=50K']['age'].std())


#6. Правда ли, что люди, которые получают больше 50k, имеют как минимум высшее образование?
# (признак education - Bachelors, Prof-school, Assoc-acdm, Assoc-voc, Masters или Doctorate)
print('#6')
n6 = pd.DataFrame(data[data['salary']== '>50K']['education'].value_counts(normalize = True))
n6.plot(kind = 'bar')
#plt.show()

#7. Выведите статистику возраста для каждой расы (признак race) и каждого пола. Используйте groupby и describe.
# Найдите таким образом максимальный возраст мужчин расы Amer-Indian-Eskimo.
print('#7')
columns_to_show = ['age']
print(data.groupby(['race','sex'])[columns_to_show].describe())

#8. Среди кого больше доля зарабатывающих много (>50K): среди женатых или холостых мужчин (признак marital-status)?
# Женатыми считаем тех, у кого marital-status начинается с Married
# (Married-civ-spouse, Married-spouse-absent или Married-AF-spouse), остальных считаем холостыми.

print('#8')
list = ['Married-civ-spouse', 'Married-spouse-absent' , 'Married-AF-spouse']
data['is_married'] = 0
data['is_married'] = (data['marital-status'].isin(list))*1
print(data.head(10))
print(pd.crosstab(data['salary'],data['is_married'], normalize = True))


#9. Какое максимальное число часов человек работает в неделю (признак hours-per-week)?
# Сколько людей работают такое количество часов и каков среди них процент зарабатывающих много?

print('#9')
print(data['hours-per-week'].max())
print(data[data['hours-per-week'] == data['hours-per-week'].max()].count(axis = 0 ))
data['hardworkers'] = 0
data['hardworkers'] = (data['hours-per-week'] == data['hours-per-week'].max())*1
print(pd.crosstab(data['salary'],data['hardworkers'],normalize = True))

#10. Посчитайте среднее время работы (hours-per-week) зарабатывающих мало и много (salary) для каждой страны (native-country).

print('#10')
columns_to_show = ['hours-per-week']
data_japan = pd.DataFrame(data[data['native-country'] == 'Japan'])
print(data.groupby(['native-country','salary'])[columns_to_show].mean())
print(data_japan.groupby(['native-country','salary'])[columns_to_show].mean())
print('#10/2')

pd.crosstab(data['native-country'], data['salary'],
           values=data['hours-per-week'], aggfunc=np.mean)
